tortoise/ninja_tortoise.go

```diff
@@ -253,7 +256,9 @@ func (ni *ninjaTortoise) updateCorrectionVectors(p votingPattern, bottomOfWindow
 func (ni *ninjaTortoise) updatePatternTally(newMinGood votingPattern, botomOfWindow types.LayerID, correctionMap map[types.BlockID]vec, effCountMap map[types.LayerID]int) {
 	ni.Debug("update tally pbase id:%d layer:%d p id:%d layer:%d", ni.pBase.id, ni.pBase.Layer(), newMinGood.id, newMinGood.Layer())
 	for idx, effc := range effCountMap {
+		ni.tGoodLock.RLock()
```

Comment:
i thought about it and since the tortoise is single threded we just need to protect the GetGoodPattern
so all the read locks except the ons in GetGoodPattern are irrelevant

Reply:
Yes, however, I rather have a safer code than a negligibile performance advantage.
If at any point we notice it would become significant we reconsider it.